weather_stats/weatherstatsapp/weather/src/Search.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from "react"
+import Temperature from "./Temperature"
+import SearchForm from "./SearchForm"
+import ResultListItemBoundary from "./ResultListItemBoundary"
+import ResultListItemPlace from "./ResultListItemPlace"
+
+const Search = () => {
+    const [searchInput, setSearchInput] = useState("")
+    const [resultList, setResultList] = useState([])
+    const [weather, setWeather] = useState(null)
+    const [selectedCity, setSelectedCity] = useState(null)
+    const [loading, setLoading] = useState(false)
+    const [weatherLoading, setWeatherLoading] = useState(false)
+    const [error, setError] = useState(false)
+
+    const handleResultClick = async (clickedData) => {
+        setWeatherLoading(true)
+        setWeather(true)
+        setError(false)
+        await new Promise((resolve) => setTimeout(resolve, 600))
```

Comment:
Oh, that's interesting. I wonder what this one aims to solve? Definitely "waits" but for what exactly?

Reply:
Just create a more smooth loading animation rather than popping up instantly and removing instantly. This is basically just UX related. 

Similar to like if apps loads too fast then users can feel it does not or did not react, but if you display a small piece of loading then it makes it seem like the app is working.